src/components/o-modal/index.js

```diff
@@ -0,0 +1,69 @@
+import PropTypes from '../../js/prop-types';
+import BaseComponentGlobal from '../../js/abstract/base-component-global';
+import defineOnce from '../../js/define-once';
+import styles from './index.scss';
+import template from './_template';
+
+class AXAModal extends BaseComponentGlobal {
+  static tagName = 'axa-modal';
+  static propTypes = {
+    open: PropTypes.bool,
+  };
+
+  set open(value) {
+    this.setAttribute('open', value);
+  }
+
+  get open() {
+    return this.getAttribute('open');
+  }
+
+  closeModal() {
+    this.removeAttribute('open');
+  }
+
+  closeModalByClickingOutside(event) {
+    if (event.target === this.querySelector('.js-modal')) this.closeModal();
+  }
+
+  closeModalOnEsc(event) {
+    const key = event.key || event.keyCode;
+    if (key === 'Escape' || key === 'Esc' || key === 27) {
+      this.closeModal();
+    }
+  }
+
+  init() {
+    super.init({ styles, template });
+  }
+
+  static get observedAttributes() {
+    return ['open'];
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    const closeModalFunction = () => this.closeModal();
+
+    const axaButtonsRaw = this.querySelectorAll('axa-button[data-modal-close]');
+    const axaButtons = Array.from(axaButtonsRaw);
+    if (axaButtons && axaButtons.length) {
+      axaButtons.forEach(button => button.addEventListener('click', closeModalFunction));
+    }
+
+    document.addEventListener('keyup', event => this.closeModalOnEsc(event));
+    this.addEventListener('click', event => this.closeModalByClickingOutside(event));
+
+    const closeButton = this.querySelector('.js-modal-close-button');
+    closeButton.addEventListener('click', closeModalFunction);
+  }
+
+  disconnectedCallback() {
+    document.removeEventListener('keyup', this.closeModalOnEsc);
```

Comment:
@raphaellueckl im not sure this can work. `this.closeModalOnEsc` is another function compared to ` event => this.closeModalOnEsc(event)`. I think they must be the same in order to be deleted.

check out here: https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener

Reply:
@AndyOGo do u know if this works?

Reply:
That creates a possible memory leak.
Agree, definitely learn how events work in the DOM.
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener

I have written utilities to make working with native events in a cross-browser manner easier, see the `on` helper:
https://github.com/axa-ch/patterns-library/blob/develop/src/js/on.js

Reply:
Cool, thanks for the feedback guys!

Reply:
Please have a look at the implementation now.